dask/backends.py

```diff
@@ -34,27 +40,27 @@ def detect_entrypoints(entry_point_name):
 BackendFuncType = TypeVar("BackendFuncType", bound=Callable[..., Any])
 
 
-class CreationDispatch:
+class CreationDispatch(Generic[BackendEntrypointType]):
     """Simple backend dispatch for collection-creation functions"""
 
-    _lookup: dict
+    _lookup: dict[str, BackendEntrypointType]
     _module_name: str
     _config_field: str
     _default: str
-    _entrypoint_class: type
+    _entrypoint_class: type[BackendEntrypointType]
 
     def __init__(
         self,
         module_name: str,
         default: str,
+        entrypoint_class: type[BackendEntrypointType],
         name: str | None = None,
-        entrypoint_class: type | None = None,
```

Comment:
I think that making this argument non-optional is the only actual runtime change here.

Reply:
Seems reasonable to me